testing/trino-testing/src/main/java/io/trino/testing/TestingSessionContext.java

```diff
@@ -29,13 +32,27 @@ private TestingSessionContext() {}
     public static SessionContext fromSession(Session session)
     {
         requireNonNull(session, "session is null");
+
+        Set<String> enabledRoles = session.getIdentity().getEnabledRoles();
+        SelectedRole selectedRole;
+        if (enabledRoles.isEmpty()) {
+            selectedRole = new SelectedRole(Type.NONE, Optional.empty());
+        }
+        else if (enabledRoles.size() == 1) {
+            selectedRole = new SelectedRole(Type.ROLE, Optional.of(enabledRoles.iterator().next()));
+        }
+        else {
+            selectedRole = new SelectedRole(Type.ALL, Optional.empty());
```

Comment:
This is a test utility to quickly convert from a Session to a SessionContext, which never happens in real code.  The problem is you can't go back from these three enum value to a list of roles, so this code assumes that if you have only one role, then we select that one, and if you have multiple roles, then that represents ALL roles you could possibly have.  This works because trino only supports no roles at all, exactly one role, or all roles you have access to.  Of course this code could convert ALL to NONE or one role, if you have no  roles or just one role, but it doesn't matter for tests.